### PR TITLE
 Consolidate test and release workflows + add concurrency limit to test + fuzz

### PR DESCRIPTION
**Pull request by ichard26**
### Description

This makes changes across workflow domains easier (like adding concurrency limits) and overall improves consistency. I plan to submit another PR integrating the mypyc workflow from ichard26/black-mypyc-wheels after this is merged. Here's an example run: https://github.com/ichard26/black/actions/runs/2144614522 (including the mypyc patch)

This is probably best landed with a rebase merge fyi.

### Checklist - did you ...

- [x] Add a CHANGELOG entry if necessary? -> n/a
- [x] Add / update tests if necessary? -> n/a
- [x] Add new / update outdated documentation?

### CODE REVIEW

**Review by cooperlees — COMMENTED** [review]
LGTM - How do we test this tho?

Want to update bandersnatch to this cause I do have a release I need to do, would rather screw up a bandersnatch release than black :D

File: docs/contributing/release_process.md [hunk 1]
```diff
@@ -88,39 +88,44 @@ Use the following template for a clean changelog after the release:
...
-This workflow builds a Python
-[sdist](https://docs.python.org/3/distutils/sourcedist.html) and
-[wheel](https://pythonwheels.com) using the latest
-[setuptools](https://pypi.org/project/setuptools/) and
-[wheel](https://pypi.org/project/wheel/) modules.
+### PyPI
 
-It will then use [twine](https://pypi.org/project/twine/) to upload both release formats
-to PyPI for general downloading of the _Black_ Python package. This is where
+PyPI is where almost all users will get _Black_ from as it is where where
```
Comment (JelleZijlstra) [inline docs/contributing/release_process.md:113]:
```suggestion
PyPI is where almost all users will get _Black_ from as it is where
```

**Comment by JelleZijlstra** [comment]
Unfortunately this has a lot of conflicts now, @ichard26 are you able to get back to it?

**Comment by ichard26** [comment]
#3189, #3190, and #3197 cover the majority of this PR. Documentation updates will come later.

**Pull request closed without merging**